File: v29_state/python/game1.py
class Character:
    def __init__(self):
        self.state = "idle"

    def walk(self):
        if self.state == "idle":
            self.state = "walking"
            print("The character is now walking.")
        else:
            print("The character cannot walk right now.")

    def run(self):
        if self.state == "walking":
            self.state = "running"
            print("The character is now running.")
        else:
            print("The character cannot run right now.")

    def jump(self):
        if self.state == "running":
            self.state = "jumping"
            print("The character is now jumping.")
        else:
            print("The character cannot jump right now.")

    def idle(self):
        self.state = "idle"
        print("The character is now idle.")


if __name__ == "__main__":
    character = Character()

    character.walk()  # The character is now walking.
    character.run()  # The character is now running.
    character.jump()  # The character is now jumping.
    character.idle()  # The character is now idle.

    character.run()  # The character cannot run right now.
    character.jump()  # The character cannot jump right now.

File: v29_state/java/Game1.java
package v29_state.java;

class Character {
    private String state = "idle";

    public void walk() {
        if (state.equals("idle")) {
            state = "walking";
            System.out.println("The character is now walking.");
        } else {
            System.out.println("The character cannot walk right now.");
        }
    }

    public void run() {
        if (state.equals("walking")) {
            state = "running";
            System.out.println("The character is now running.");
        } else {
            System.out.println("The character cannot run right now.");
        }
    }

    public void jump() {
        if (state.equals("running")) {
            state = "jumping";
            System.out.println("The character is now jumping.");
        } else {
            System.out.println("The character cannot jump right now.");
        }
    }

    public void idle() {
        state = "idle";
        System.out.println("The character is now idle.");
    }
}

public class Game1 {
    public static void main(String[] args) {
        Character character = new Character();

        character.walk(); // The character is now walking.
        character.run(); // The character is now running.
        character.jump(); // The character is now jumping.
        character.idle(); // The character is now idle.

        character.run(); // The character cannot run right now.
        character.jump(); // The character cannot jump right now.
    }
}
